Mathlib/Geometry/Manifold/ContMDiff.lean

```diff
@@ -2030,6 +2030,83 @@ nonrec theorem Smooth.smul {f : M → 𝕜} {g : M → V} (hf : Smooth I 𝓘(
   hf.smul hg
 #align smooth.smul Smooth.smul
 
+/-! ### Smoothness of an open embedding and its inverse -/
+
+variable (I)
+lemma contMDiff_openEmbedding
+    {M : Type*} [TopologicalSpace M] [Nonempty M]
+  {e : M → H} (h : OpenEmbedding e) {n : WithTop ℕ} :
+  @ContMDiff _ _ _ _ _ _ _ I _ _ h.singletonChartedSpace _ _ _ _ _ I _ _ _ n e := by
```

Comment:
I have the nagging feeling there should be a nicer way to write this than ~a dozen underscores (say, using Lean4's named arguments). I fail to find one, though. Perhaps a reviewer has a better idea --- or you already found the best way :shrug: .

Reply:
This probably requires some changes in the `ContMDiff` API, so we probably shouldn't touch it in this PR. I suppose one could give names to the type class argument `[CS : ChartedSpace H M]` so Lean4's named argument works.